### DISCUSSION TITLE
what's the real issue for this error message: "Message is not authenticated."

### DISCUSSION THREAD

**Discussion by JackWang8202** [openpgpjs/openpgpjs#1418] [Q&A] [answered]
we got this error "Message is not authenticated." from our app while using this lib to decrypt the message. the message was signed and encrypted, can you tell what's the possible root cause for this error?
from your code, I saw this, I believe this is where the error message from:
```
// If MDC errors are not being ignored, all missing MDC packets in symmetrically encrypted data should throw an error
    if (!config.allowUnauthenticatedMessages) {
      throw new Error('Message is not authenticated.');
    }
```
but I don't understand here, why do we only check the configuration here but not the message self?

**Comment by larabr** [comment, marked as answer]
If that's where the error was thrown, it means the message does not include a protection mechanism to determine the authenticity of the decrypted data. This kind of messages [will be deprecated in the next version of OpenPGP](https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-rfc4880bis-05#section-5.8)  due to the [EFAIL attack](https://efail.de/#mitigations), so we prevent decrypting them, unless you give explicit permission by setting the config.
